### DISCUSSION TITLE
Use of the 2nd output of a chiller

### DISCUSSION THREAD

**Discussion by JulianfromBODE** [SESMG/SESMG#228] [Q&A]
Is it possible to use output 2 of a chiller? In my energy system I have chillers which use the waste heat from cooling production by means of heat recovery.  I would like to use this amount of heat in my heat_bus. Unfortunately, when a bus is connected to output 2, it is not integrated.

**Comment by jtock** [comment]
Hi @JulianfromBODE,

sorry for my late response!
I checked your problem. Right now it is not possible to have two outputs on a CompressionHeatTransformer. 
Although it is possible for some other kinds of transformers like a CHP or a generic transformer. They all have in common that they have multiple fixed efficencies. That is not possible for a CompressionHeatTransformer like heat pump or chiller, since the COP is getting calculated for each time step and is based on the given ambient temperature.
See e.g. here: https://spreadsheet-energy-system-model-generator.readthedocs.io/en/latest/02.02.00_application.html#generic-transformer

I will create an issue for that since it would be a nice feature. Unfortunately, we do not have the capacity right now to implement it soon.

If you can imagine to implement it by yourself, we can discuss a possible solution and merge it in the SESMG code. We would try to support you doing so. 
You can finde the code here for a first view: https://spreadsheet-energy-system-model-generator.readthedocs.io/en/latest/_modules/program_files/preprocessing/components/Transformer.html#

Grettings 
Jan

**Reply by JulianfromBODE** [reply]
Hi @jtock 
thanks for your detailed answer, too bad it doesn't work yet but the feature would be really cool. Until then I have found a workaround. This only works with predefined coolingsinks under time series but maybe it is still interesting for someone.

Workaround: 
Instead of the CompressionHeatTransformer I use a generic transformer and enter a heating and cooling COP (annual average value) in efficencie1 and effencie2, even if the documentation says that only values between 0 and 1 can be used here. It is possible to enter values greater than 1 (for example COPs), which are then simply used as a factor. 

Here is an example of how I created my chiller:
![image](https://github.com/SESMG/SESMG/assets/156187520/c9787213-cfc8-4175-aeba-a6ac426f4ccd)

Do you think this approach could cause any problems? 

Greetings 
Julian

**Reply by jtock** [reply]
Thank you for your advise concerning efficiencies > 1. We did not have the use-case yet, I will add it to the docu!

Your workaround should not cause any problems. You can use it like this. It just does not use the full potential of energy system modeling, since the calculation of COP values for each time step is more accurate.
